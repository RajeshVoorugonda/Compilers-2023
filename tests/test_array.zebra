zout("testcase 1 for arrays");
array int a = [1,2,3];
zout(length(a));
append(2, a);
zout(length(a));
remove(1, a);
zout(length(a));
zout(a);
zout(slice a 1:3);
zout(index a 2);
int f=length(a)-1;
zout(f);
int d=index a 1;
zout(d);
remove(f,a);
zout(a);
int g=length([1,2,3,4]);
zout(g);

remove(length(a)-1, a);
zout(a);

pop(a);
zout(a);

append(1,a);
append(2,a);
append(3,a);

array int li = [4,5,6];
zout(a+li);
array array int p = [[1],[2],[3]];
zout(p);
array int zaa = [3,6];
zout("testcase 2 for nested arrays");

array array int b = [[100],[200],[300]];
zout(length(b));
append([2], b);
zout(length(b));
remove(2, b);
zout(length(b));
zout(b);
array array int lii = [[4],[5],[6]];
zout(b+lii);


zout("testcase 3 for strings concatenation");

string ma="Fasak Ayyindi";
zout(length(ma));
string sayhi = " Hi";
string mama=ma+sayhi;
zout(length(mama));
string mamama=ma+" Bye";
zout(length(mamama));
ma=ma+" Good Bye";
zout(ma);
string r=slice ma 2:3;
zout(r);
