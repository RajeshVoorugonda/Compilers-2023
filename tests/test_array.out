testcase 1 for arrays
3
4
3
[1, 3, 2]
[3, 2]
2
2
3
[1, 3]
4
[1]
[]
[1, 2, 3, 4, 5, 6]
[[1], [2], [3]]
testcase 2 for nested arrays
3
4
3
[[100], [200], [2]]
[[100], [200], [2], [4], [5], [6]]
testcase 3 for strings concatenation
13
16
17
Fasak Ayyindi Good Bye
s